<ui version="4.0" >
 <class>JobsBase</class>
 <widget class="QWidget" name="JobsBase" >
  <property name="geometry" >
   <rect>
    <x>0</x>
    <y>0</y>
    <width>400</width>
    <height>423</height>
   </rect>
  </property>
  <property name="windowTitle" >
   <string>ThreadWeaver Jobs Examples</string>
  </property>
  <property name="toolTip" >
   <string>The program executes 100 jobs in 4 threads. Each job waits for a random number of milliseconds between 1 and 1000.</string>
  </property>
  <layout class="QGridLayout" >
   <property name="margin" >
    <number>8</number>
   </property>
   <property name="spacing" >
    <number>6</number>
   </property>
   <item row="3" column="0" colspan="5" >
    <layout class="QHBoxLayout" >
     <property name="margin" >
      <number>0</number>
     </property>
     <property name="spacing" >
      <number>6</number>
     </property>
     <item>
      <widget class="QCheckBox" name="cbLog" >
       <property name="toolTip" >
        <string>Check to see logging information about thread activity. Watch the console output to see the log information.</string>
       </property>
       <property name="text" >
        <string>Log thread activity</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer>
       <property name="orientation" >
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" >
        <size>
         <width>31</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="ThreadWeaver::WeaverThreadGrid" name="threadGrid" >
       <property name="toolTip" >
        <string>Displays Thread Activity</string>
       </property>
       <property name="frameShape" >
        <enum>QFrame::NoFrame</enum>
       </property>
       <property name="frameShadow" >
        <enum>QFrame::Plain</enum>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="6" column="0" >
    <widget class="QPushButton" name="pbStart" >
     <property name="enabled" >
      <bool>false</bool>
     </property>
     <property name="text" >
      <string>Start</string>
     </property>
    </widget>
   </item>
   <item row="6" column="1" >
    <widget class="QPushButton" name="pbStop" >
     <property name="enabled" >
      <bool>false</bool>
     </property>
     <property name="text" >
      <string>Stop</string>
     </property>
    </widget>
   </item>
   <item row="6" column="4" >
    <widget class="QPushButton" name="pbQuit" >
     <property name="enabled" >
      <bool>false</bool>
     </property>
     <property name="text" >
      <string>Quit</string>
     </property>
    </widget>
   </item>
   <item row="1" column="4" >
    <widget class="QLCDNumber" name="lcdNumJobsRem" >
     <property name="minimumSize" >
      <size>
       <width>32</width>
       <height>48</height>
      </size>
     </property>
     <property name="font" >
      <font>
       <family>Sans Serif</family>
       <pointsize>14</pointsize>
       <weight>50</weight>
       <italic>false</italic>
       <bold>false</bold>
       <underline>false</underline>
       <strikeout>false</strikeout>
      </font>
     </property>
     <property name="frameShape" >
      <enum>QFrame::NoFrame</enum>
     </property>
     <property name="frameShadow" >
      <enum>QFrame::Plain</enum>
     </property>
     <property name="segmentStyle" >
      <enum>QLCDNumber::Flat</enum>
     </property>
    </widget>
   </item>
   <item row="1" column="3" >
    <spacer>
     <property name="orientation" >
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="sizeHint" >
      <size>
       <width>40</width>
       <height>20</height>
      </size>
     </property>
    </spacer>
   </item>
   <item row="5" column="0" colspan="5" >
    <widget class="QFrame" name="horizontalLine" >
     <property name="frameShape" >
      <enum>QFrame::HLine</enum>
     </property>
     <property name="frameShadow" >
      <enum>QFrame::Sunken</enum>
     </property>
    </widget>
   </item>
   <item row="6" column="2" colspan="2" >
    <spacer>
     <property name="orientation" >
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="sizeHint" >
      <size>
       <width>136</width>
       <height>20</height>
      </size>
     </property>
    </spacer>
   </item>
   <item row="0" column="0" colspan="5" >
    <widget class="QLabel" name="label" >
     <property name="cursor" >
      <cursor>4</cursor>
     </property>
     <property name="frameShape" >
      <enum>QFrame::Panel</enum>
     </property>
     <property name="frameShadow" >
      <enum>QFrame::Sunken</enum>
     </property>
     <property name="lineWidth" >
      <number>2</number>
     </property>
     <property name="text" >
      <string>GUI based example for the Weaver Thread Manager</string>
     </property>
     <property name="wordWrap" >
      <bool>true</bool>
     </property>
     <property name="margin" >
      <number>6</number>
     </property>
    </widget>
   </item>
   <item row="2" column="0" colspan="5" >
    <widget class="QProgressBar" name="pbProgress" >
     <property name="value" >
      <number>0</number>
     </property>
     <property name="textVisible" >
      <bool>false</bool>
     </property>
    </widget>
   </item>
   <item row="1" column="0" colspan="3" >
    <widget class="QLabel" name="label_2" >
     <property name="text" >
      <string>Remaining number of jobs:</string>
     </property>
    </widget>
   </item>
   <item row="4" column="1" >
    <spacer>
     <property name="orientation" >
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" >
      <size>
       <width>85</width>
       <height>101</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>ThreadWeaver::WeaverThreadGrid</class>
   <extends>QFrame</extends>
   <header>WeaverThreadGrid.h</header>
   <container>0</container>
   <pixmap></pixmap>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>pbStart</tabstop>
  <tabstop>pbStop</tabstop>
  <tabstop>pbQuit</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
